streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamTransformValuesTest.java

```diff
@@ -138,6 +138,54 @@ public void close() { }
         assertArrayEquals(expected, supplier.theCapturedProcessor().processed.toArray());
     }
 
+    @Test
+    public void shouldEmitNoRecordIfTransformReturnsNull() {
```

Comment:
req: I guess, I was not clear enough in my previous comment. Sorry for that! I see that the other unit tests in `KStreamTransformValuesTest` are also `TopologyTestDriver` tests. However, I think that is not needed in this case. `TopologyTestDriver` tests are harder to maintain and slower to run as a simple unit test like `KStreamFlatTransformValuesTest#shouldEmitNoRecordIfTransformReturnsNull()`. We need to rewrite `TopologyTestDriver` tests in `KStreamTransformValueTest` to simple unit tests at some point (not in this PR). Please use a simple unit test similar to the following.
```
    @Test
    public void shouldEmitNoRecordIfTransformReturnsNull() {
        final ProcessorContext context = mock(ProcessorContext.class);
        final ValueTransformerWithKey<Integer, Integer, Iterable<String>> valueTransformer = mock(ValueTransformerWithKey.class);
        EasyMock.expect(valueTransformer.transform(inputKey, inputValue)).andReturn(null);
        EasyMock.replay(context, valueTransfomer);
        final Integer inputKey = 1;
        final Integer inputValue = 10;
        final KStreamFlatTransformValuesProcessor<Integer, Integer, String> processor = new KStreamFlatTransformValuesProcessor<>(valueTransformer);
        processor.init(context);

        processor.process(inputKey, inputValue);

        EasyMock.verify(context, valueTransformer);
    }
```

Reply:
Not a problem. Updated.